### DISCUSSION TITLE
Track torrent download progress excluding any files with 0 priority

### DISCUSSION THREAD

**Discussion by xob0t** [rmartin16/qbittorrent-api#386] [Q&A] [answered]
I'm having trouble finding an approach to get a completion percentage for a torrent in which some files' priority is set to 0.
I want to track progress excluding any files with 0 priority.

**Comment by rmartin16** [comment, marked as answer]
The torrent [dictionaries](https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)#get-torrent-list) returned from `client.torrents.info()` contain a `size` element that represents the total amount of data to be downloaded. The `completed` element of that dictionary contains how much has been downloaded. The ratio of those two numbers is the percentage complete for the torrent download.

**Comment by xob0t** [comment]
My problem is that if torrent has two files with equal size, and i set one's priority to 0 and download the other, the total progress would still be 50% even though i don't need the other file.
I'm thinking we can sum all sizes of any files with priority above 0, then ratio downloaded bytes with it. But it seem like on completion, that ratio would usually be bigger than 1, as files and chunks are not aligned.
In that case, i can visualize progress that way, but to reliably decide when the torrent is complete i need to check its state, or maybe check every file's status?

**Reply by rmartin16** [reply]
Have you tried what I wrote?

I implemented this is [qbittorrentui](https://github.com/rmartin16/qbittorrentui); the values for `size` and `completed` already take in to account which pieces of content are selected for download. And yes; there are the edge cases of whether the torrent has started or already finished you'll need to account for.

**Reply by xob0t** [reply]
Oh, I must have misread your message :)
In that case, that is all I need. Thank you for your help, fast responses, and that awesome, easy-to-use library!
